### ISSUE TITLE
Add updated SPHARM-PDM functionality to SlicerSALT build 

### ISSUE THREAD

**Issue by bpaniagua** [Kitware/SlicerSALT#306]
https://github.com/Kitware/SlicerSALT/blob/master/CMakeLists.txt#L399

@allemangD could you please link here the SPHARM-PDM fork that includes your WIP on the python implementation of SPHARM-PDM?

---

Plan is to include the rework version as an "experimental" alternative for SPHARM-PDM.

**Comment by allemangD** [comment]
https://github.com/slicersalt/py-spharm-pdm/

Two attempts at optimization with branches `torch` and `torch-optim` - neither work very well for one reason or another.

Currently the two core challenges:

1. `scipy.optimize` uses dense matrices for storing Jacobian and Hessian information, so its constrained optimizers use a prohibitive amount of memory.
  - The branch `torch` attempts to use pytorch backward autograd to compute these instead. Runtime improves but memory usage still prohibitive.
  - The branch `torch-optim` attempts to use `torch.optim` library, however it doesn't support constrained optimizers. I attempted a constraint penalty term, but it doesn't perform well: it optimizes for the constraint _or_ the metric, not 
2. There is a bug somewhere in `core.initial_parameterization` that causes self-intersecting initial parameterization; since the initial conditions are wrong the optimizer fails to produce any meaningful result.

The problem with the python optimizers is we need _sparse and constrained optimization_. I'm curious to investigate the package [pyOptSparse](https://github.com/mdolab/pyoptsparse/tree/main); it claims to support this. It doesn't provide an autograd interface but I can attempt with the same technique as in the `torch` branch.

The first issue to resolve is the discrepency in initial parameterization. My current Python implementation is based on the paper, not the code. I've isolated the relevant sections of `EqualAreaParametricMeshNewtonIterator` and am comparing them now. I will follow up here with details, or schedule a breakout with you and jared if the bug is too subtle.

Note for posterity: the relevant methods in `EqualAreaParametricMeshNewtonIterator` are:

- `start_values` invokes general workflow.
- `generate_matrix` populates laplacian matrix for latitude problem.
- `set_lati_rhs` populates rhs vector for latitude problem.
- Solve `Ax = b`; extract latitudes from `x`.
- `modify_matrix` updates the matrix for the longitude problem.
- `set_longi_rhs` populates rhs vector for longitude problem.
- Solve `Ax = b`; extract longitudes from `x`.
- Compute cartesian coordinates.

**Comment by bpaniagua** [comment]
And associated blog post.

**Comment by bpaniagua** [comment]
Blog published 
https://www.kitware.com/?p=77645&preview=true&_thumbnail_id=77652
